### DISCUSSION TITLE
Custom success verification message

### DISCUSSION THREAD

**Discussion by AlexTsIvanov** [pocketbase/pocketbase#1088] [Q&A] [answered]
Hey,
Is there a way to have a custom message and design when user clicks the verify email link or better some redirectUrl the user is sent to.

Also, another question. I want to update some field once the user verified his email (basically want to add free credits for verifying account). I couldn't see an easy way to do that. Any ideas?

**Comment by ganigeorgiev** [comment, marked as answer]
> Is there a way to have a custom message and design when user clicks the verify email link or better some redirectUrl the user is sent to.

You can provide a custom verification confirm url from the "Admin UI > Settings > Mail settings > Default Verification email template" and change the "Action URL" field.

In your custom endpoint, then you'll have to call the `POST /api/collections/users/confirm-verification` endpoint with the token from the action url. You can also use the SDKs:

```js
import PocketBase from 'pocketbase';

const pb = new PocketBase('https://pocketbase.io');

await pb.collection('users').confirmVerification('TOKEN');
```
_(more info you could find in the API Preview for your `users` collection)_

---

> I want to update some field once the user verified his email (basically want to add free credits for verifying account). I couldn't see an easy way to do that. Any ideas?

This could be done either client-side, eg. after the confirmVerification call, by calling `update` on the user model OR use the [`OnModelBeforeUpdate` Go event hook](https://pocketbase.io/docs/event-hooks/#db-hooks):

```go
app.OnModelBeforeUpdate().Add(func(e *core.ModelEvent) error {
	record, ok := e.Model.(*models.Record)
	if !ok || record.Collection().Name != "users" {
		return nil // not a user record model
	}

	// in the future the old state will be part of the event
	oldRecord, err := app.Dao().FindRecordById("users", record.Id)
	if err != nil {
		return err
	}

	if record.Verified() && oldRecord.Verified() != record.Verified() {
		// overwrite/set field(s)
		// note: you don't need to explicitly "save" the record
		// 	     because the finalizer of the hook will do it for us if no error is returned by all listeners.
		record.Set("yourField", "test")
	}

	return nil
})
```
_(in the near future there will dedicated event hooks for before/after verification per https://github.com/pocketbase/pocketbase/issues/468)_
